### ISSUE TITLE
IDataConverter.ToData() method signature doesn't match GOLANG or Java

### ISSUE THREAD

**Issue by jefflill** [nforgeio/neonKUBE#797]
The .NET client's `IDataConverter.ToData()` interface method accepts only a single object parameter whereas the Java and GOLANG clients accept a variable number of parameters.  We need to modify the .NET client to match.  This is related to #783.

Here's the Java source code comment describing how this is encoded:
```
/**
 * When values is empty or it contains a single value and it is null then return empty blob. If a
 * single value do not wrap it into Json array. Exception stack traces are converted to a single
 * string stack trace to save space and make them more readable.
 */
```
We're going to ignore the stack trace part for now.

**Comment by jefflill** [comment]
After looking throughout code, I see that we're already handling this logic in via the `CadenceHelper.ArgsToBytes()` and `BytesToArgs()` methods and that these methods also implement backwards compatibility for #783 so I don't want to replace those.

I do believe we that we can also update the `IDataConverter.ToData()` method to match the other clients without messing with the argument related methods above.

We also need to document this behavior so that anybody implementing a custom converter will know what they need to do.

**Comment by jefflill** [comment]
**FIXED** for v2.0.0

**Comment by jefflill** [comment]
**DONE**
